src/main/java/com/vaadin/hackathon/git/GitHubTagService.java

```diff
@@ -0,0 +1,196 @@
+                    if (createdBy == null) createdBy = author.path("name").asText();
+                    createdAt = target.path("committedDate").asText();
+                }
+                // System.out.printf("Tag: %s, Created by: %s, At: %s%n", tagName, createdBy, createdAt);
+                result.add(new VersionDetails(tagName, OffsetDateTime.parse(createdAt) ,createdBy));
+            }
+
+            // Handle pagination
+            JsonNode pageInfo = refs.path("pageInfo");
+            boolean hasNext = pageInfo.path("hasNextPage").asBoolean();
+            cursor = hasNext ? pageInfo.path("endCursor").asText() : null;
+        } while (cursor != null);
+
+        saveToCache(repoOwner, repoName, result);
+        return result;
+    }
+
+    private static JsonNode fetchTags(String owner, String name, int pageSize, String after, String githubToken) throws IOException, InterruptedException {
+        // Build GraphQL variables
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new JavaTimeModule());
+        ObjectNode variables = mapper.createObjectNode()
+                .put("owner", owner)
+                .put("name", name)
+                .put("pageSize", pageSize);
+        if (after != null) variables.put("after", after);
+
+        ObjectNode payload = mapper.createObjectNode();
+        payload.put("query", QUERY);
+        payload.set("variables", variables);
+        String requestBody = payload.toString();
+
+        var builder = HttpRequest.newBuilder()
+                .uri(URI.create(GITHUB_API_URL))
+                .timeout(Duration.ofSeconds(60))
+                .header("Content-Type", "application/json")
+                .POST(HttpRequest.BodyPublishers.ofString(requestBody));
+
+        if(githubToken!=null && !githubToken.isEmpty()){
+            builder.header("Authorization", "Bearer " + githubToken);
+        }else{
+            LOGGER.info("Querying API without bearer token. You might run into rate limitations.");
+        }
+
+        HttpRequest request = builder.build();
+
+        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        if (response.statusCode() == 403 || response.statusCode() == 429){
+            throw new IOException("GraphQL API quota exceeded!");
+        }else if (response.statusCode() / 100 != 2) {
+            throw new IOException("Unexpected response status: " + response.statusCode() + " body: " + response.body());
+        }
+
+        JsonNode root = mapper.readTree(response.body());
+        if (root.has("errors")) {
+            throw new IOException("GraphQL errors: " + root.path("errors").toString());
+        }
+        return root.path("data");
+    }
+
+    private static List<VersionDetails> fetchTagsFromCache(String repoOwner, String repoName) {
+        try {
+            File cacheFile = new File(System.getProperty("java.io.tmpdir"), 
```

Comment:
instead of this tmp folder, make it configurable through a variable in application.properties